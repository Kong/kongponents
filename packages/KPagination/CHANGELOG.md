# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 6.1.3 (2021-11-17)

**Note:** Version bump only for package @kongponents/kpagination
