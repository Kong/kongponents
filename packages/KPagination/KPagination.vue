<template>
  <nav
    aria-label="Pagination Navigation"
    data-testid="k-pagination-container">
    <div
      v-if="totalCount > currentPageSize"
      class="card-pagination-bar">
      <span
        class="pagination-text"
        data-testid="visible-items">
        <span class="pagination-text-pages">{{ pagesString }}</span>
        {{ pageCountString }}
      </span>
      <ul class="pagination-button-container">
        <li
          :class="{ disabled: backDisabled }"
          class="pagination-button square"
          data-testid="prev-btn">
          <a
            href="#"
            aria-label="Go to the previous page"
            @click.prevent="pageBack">
            <KIcon
              :color="backDisabled ? 'var(--grey-500)' : 'var(--blue-400)'"
              icon="arrowLeft"
              width="16px"
              heigth="16px"
              view-box="0 0 16 14"
            />
          </a>
        </li>
        <li
          v-if="firstDetached"
          class="pagination-button"
          data-testid="page-1-btn">
          <a
            href="#"
            aria-label="Go to the first page"
            @click.prevent="changePage(1)">
            1
          </a>
        </li>
        <li
          v-if="firstDetached"
          class="pagination-button placeholder">
          ...
        </li>
        <li
          v-for="page in pagesVisible"
          :key="page"
          :class="{ active: page == currentlySelectedPage }"
          :data-testid="`page-${page}-btn`"
          class="pagination-button">
          <a
            :aria-label="`Go to page ${page}`"
            :aria-current="page == currentlySelectedPage && 'page'"
            href="#"
            @click.prevent="changePage(page)">
            {{ page }}
          </a>
        </li>
        <li
          v-if="lastDetached"
          class="pagination-button placeholder">
          ...
        </li>
        <li
          v-if="lastDetached"
          class="pagination-button">
          <a
            href="#"
            aria-label="Go to the last page"
            data-testid="last-btn"
            @click.prevent="changePage(pageCount)">
            {{ pageCount }}
          </a>
        </li>
        <li
          :class="{ disabled: forwardDisabled }"
          class="pagination-button square"
          data-testid="next-btn">
          <a
            href="#"
            aria-label="Go to the next page"
            @click.prevent="pageForward">
            <KIcon
              :color="forwardDisabled ? 'var(--grey-500)' : 'var(--blue-400)'"
              icon="arrowRight"
              width="16px"
              heigth="16px"
              view-box="0 0 16 14"
            />
          </a>
        </li>
      </ul>
      <span
        class="page-size-select"
        data-testid="page-size-dropdown">
        <KSelect
          :placeholder="`${currentPageSize} items per page`"
          :items="pageSizeOptions"
          :test-mode="testMode"
          :button-text="pageSizeText"
          :kpop-attributes="kpopAttrs"
          appearance="button"
          @selected="updatePageSize"
        />
      </span>
    </div>
  </nav>
</template>

<script>
import KIcon from '@kongponents/kicon/KIcon.vue'
import KSelect from '@kongponents/kselect/KSelect.vue'

export default {
  name: 'KPagination',
  components: {
    KIcon,
    KSelect
  },
  props: {
    items: {
      type: Array,
      default: () => []
    },
    totalCount: {
      type: Number,
      default: 0
    },
    pageSizes: {
      type: Array,
      default: () => [15, 25, 50, 75, 100]
    },
    neighbors: {
      type: Number,
      default: 1
    },
    searchTriggered: {
      type: Boolean,
      default: false
    },
    currentPage: {
      type: Number,
      default: null
    },
    /**
     * Test mode - for testing only, strips out generated ids
     */
    testMode: {
      type: Boolean,
      default: false
    }
  },
  data () {
    const currPage = this.currentPage ? this.currentPage : 1
    const currentPageSize = this.pageSizes[0]
    const pageCount = Math.ceil(this.totalCount / currentPageSize)

    const pageSizeOptions = this.pageSizes.map((size, i) => ({
      label: `${size}`,
      key: `size-${i}`,
      value: size
    }))

    return {
      kpopAttrs: {
        placement: 'top'
      },
      currPage,
      currentPageSize,
      pageCount,
      pageSizeOptions,
      backDisabled: currPage === 1,
      forwardDisabled: this.totalCount === 1,
      pageSizeText: '',
      pagesVisible: this.getVisiblePages(
        this.currentlySelectedPage,
        pageCount,
        false,
        pageCount > 5 + 2 * this.neighbors
      ),
      firstDetached: false,
      lastDetached: pageCount > 5 + 2 * this.neighbors
    }
  },
  computed: {
    startCount () {
      return (this.currPage - 1) * this.currentPageSize + 1
    },
    endCount () {
      let calculatedEndCount = this.startCount - 1 + this.currentPageSize

      return calculatedEndCount > this.totalCount
        ? this.totalCount
        : calculatedEndCount
    },
    pagesString () {
      return `${this.startCount} to ${this.endCount}`
    },
    pageCountString () {
      return ` of ${this.totalCount}`
    },
    currentlySelectedPage () {
      if (this.currentPage) {
        return this.currentPage
      } else {
        return this.currPage
      }
    }
  },
  watch: {
    searchTriggered (newval) {
      if (newval === true) {
        this.returnToFirstPage()
      }
    }
  },
  methods: {
    pageForward () {
      if (this.forwardDisabled) {
        return
      }

      this.currPage++
      this.updatePage()
    },
    pageBack () {
      if (this.backDisabled) {
        return
      }

      this.currPage--
      this.updatePage()
    },
    changePage (page) {
      this.currPage = page
      this.updatePage()
    },
    updatePage () {
      const lastEntry =
        (this.currPage - 1) * this.currentPageSize + this.currentPageSize

      this.forwardDisabled = lastEntry >= this.totalCount
      this.backDisabled = this.currPage === 1

      // The view will hold
      // Selected page, first page, last page, 2 placeholders and 2 * neighbors
      const visiblePages = 5 + 2 * this.neighbors

      if (this.pageCount <= visiblePages) {
        // All pages will fit in screen
        this.firstDetached = false
        this.lastDetached = false
      } else {
        this.firstDetached = this.currPage >= this.neighbors + 4
        this.lastDetached =
          this.currPage <= this.pageCount - this.neighbors - 3
      }

      this.pagesVisible = this.getVisiblePages()

      this.$emit('pageChanged', {
        page: this.currPage,
        pageCount: this.pageCount,
        firstItem: this.startCount,
        lastItem: this.endCount,
        visibleItems: this.items.slice(this.startCount - 1, this.endCount)
      })
    },
    updatePageSize (event) {
      this.currentPageSize = event.value
      this.pageSizeText = this.currentPageSize + ' items per page'
      this.pageCount = Math.ceil(this.totalCount / this.currentPageSize)

      this.$emit('pageSizeChanged', {
        pageSize: this.currentPageSize,
        pageCount: this.pageCount
      })

      this.changePage(1)
    },
    getVisiblePages (
      currPage = this.currPage,
      pageCount = this.pageCount,
      firstDetached = this.firstDetached,
      lastDetached = this.lastDetached
    ) {
      let pages = [...Array(pageCount).keys()].map((n) => n + 1)

      const visiblePages = 5 + 2 * this.neighbors
      // All pages fit on one screen
      if (pages.length <= visiblePages) {
        return pages
      }

      if (!firstDetached) {
        // First pages
        pages = pages.filter((n) => n <= this.neighbors * 2 + 3)
      } else if (firstDetached && lastDetached) {
        // Middle pages (if they do not fit on one screen)
        pages = pages.filter(
          (n) =>
            n > currPage - this.neighbors - 1 &&
            n < currPage + this.neighbors + 1
        )
      } else if (firstDetached && !lastDetached) {
        // Last pages
        pages = pages.filter((n) => n > pageCount - this.neighbors * 2 - 3)
      }

      return pages
    }
  }
}
</script>

<style lang="scss" scoped>
.card-pagination-bar {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.pagination-text {
  font-size: 13px;
  color: var(--grey-500);
  &-pages {
    color: black;
  }
}

.page-size-select {
  --KButtonFontSize: var(--type-sm);
  color: var(--blue-400);
  font-weight: 500;
  line-height: 20px;
}

.pagination-button-container {
  display: flex;
  list-style: none;
  text-align: center;

  a {
    text-decoration: none !important;
    font-weight: initial;
    display: block;
  }

  .pagination-button {
    align-self: center;
    width: 32px;
    height: 32px;

    line-height: 32px;
    font-weight: initial;
    color: var(--grey-500);
    border: 1px solid var(--grey-200);
    background-color: white;
    border-radius: 4px;
    margin: 0 6px;
    cursor: pointer;

    &.square {
      border: none;
    }

    &.placeholder {
      cursor: initial;
    }

    &:focus:not(.placeholder),
    &:hover:not(.placeholder) {
      color: var(--blue-500);
      border-color: var(--blue-500);
      border-radius: 4px;
    }
    &.disabled:focus:not(.placeholder),
    &.disabled:hover:not(.placeholder) {
      color: var(--black-45);
      border-color: var(--grey-200);
    }

    &.active {
      outline: none;
      color: var(--blue-500);
      border-color: var(--blue-200);
      border-radius: 4px;
      background-color: var(--blue-100);
    }
  }
}
</style>

<style lang="scss">
.page-size-select {
  .k-select-pop-button[x-placement^="top"] {
    margin-bottom: 2px;
  }
}
</style>
