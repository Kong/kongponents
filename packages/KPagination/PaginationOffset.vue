<template>
  <ul class="pagination-button-container mb-0 pa-0">
    <li
      :class="{ disabled: prevButtonDisabled }"
      class="pagination-button"
      data-testid="prev-btn">
      <a
        href="#"
        aria-label="Go to the previous page"
        @click.prevent="getPrevOffset">
        <KIcon
          :color="prevButtonDisabled ? 'var(--grey-500)' : 'var(--blue-400)'"
          icon="arrowLeft"
          size="16"
          view-box="0 0 16 14"
        />
      </a>
    </li>
    <li
      :class="{ disabled: nextButtonDisabled }"
      class="pagination-button"
      data-testid="next-btn">
      <a
        href="#"
        aria-label="Go to the next page"
        @click.prevent="getNextOffset">
        <KIcon
          :color="nextButtonDisabled ? 'var(--grey-500)' : 'var(--blue-400)'"
          icon="arrowRight"
          size="16"
          view-box="0 0 16 14"
        />
      </a>
    </li>
  </ul>
</template>

<script>
import KIcon from '@kongponents/kicon/KIcon.vue'
import KSelect from '@kongponents/kselect/KSelect.vue'

export default {
  name: 'PaginationOffset',
  components: {
    KIcon,
    KSelect
  },
  props: {
    prevButtonDisabled: {
      type: Boolean,
      default: false
    },
    nextButtonDisabled: {
      type: Boolean,
      default: false
    }
  },
  methods: {
    getNextOffset () {
      if (this.nextButtonDisabled) {
        return
      }

      this.$emit('getNextOffset')
    },
    getPrevOffset () {
      if (this.prevButtonDisabled) {
        return
      }

      this.$emit('getPrevOffset')
    }
  }
}
</script>
