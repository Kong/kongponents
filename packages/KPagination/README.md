# @kongponents/kpagination

[![](https://img.shields.io/npm/v/@kongponents/kpagination.svg?style=flat-square)](https://www.npmjs.com/package/@kongponents/kpagination)

```vue
<KPagination :description="'hello world'">
  Hello from a slot
</KPagination>
```
