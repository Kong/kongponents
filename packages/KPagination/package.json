{
  "name": "@kongponents/kpagination",
  "version": "6.7.8",
  "description": "Pagination component",
  "main": "dist/KPagination.umd.min.js",
  "componentName": "KPagination",
  "source": "KPagination.vue",
  "files": [
    "dist",
    "KPagination.vue"
  ],
  "repository": {
    "type": "git",
    "url": "git+https://github.com/Kong/kongponents.git"
  },
  "author": "Kong Inc",
  "license": "Apache-2.0",
  "bugs": {
    "url": "https://github.com/Kong/kongponents/issues"
  },
  "homepage": "https://github.com/Kong/kongponents#readme",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@kongponents/kicon": "^6.7.8",
    "@kongponents/kselect": "^6.7.8",
    "@kongponents/styles": "^6.7.8"
  }
}
