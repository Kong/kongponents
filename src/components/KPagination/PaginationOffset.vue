<template>
  <div class="pagination-offset-button-container mb-0 pa-0">
    <KButton
      :class="{ disabled: prevButtonDisabled }"
      class="pagination-button"
      data-testid="prev-btn"
      aria-label="Go to the previous page"
      @click.prevent="getPrevOffset"
    >
      <template #icon>
        <KIcon
          :color="prevButtonDisabled ? 'var(--grey-500)' : 'var(--blue-400)'"
          icon="arrowLeft"
          size="16"
          view-box="0 0 16 16"
        />
      </template>
    </KButton>
    <KButton
      :class="{ disabled: nextButtonDisabled }"
      class="pagination-button"
      data-testid="next-btn"
      aria-label="Go to the next page"
      @click.prevent="getNextOffset"
    >
      <template #icon>
        <KIcon
          :color="nextButtonDisabled ? 'var(--grey-500)' : 'var(--blue-400)'"
          icon="arrowRight"
          size="16"
          view-box="0 0 16 16"
        />
      </template>
    </KButton>
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue'
import KIcon from '@/components/KIcon/KIcon.vue'

export default defineComponent({
  name: 'PaginationOffset',
  components: {
    KIcon,
  },
  props: {
    prevButtonDisabled: {
      type: Boolean,
      default: false,
    },
    nextButtonDisabled: {
      type: Boolean,
      default: false,
    },
    /**
     * Test mode - for testing only, strips out generated ids
     */
    testMode: {
      type: Boolean,
      default: false,
    },
  },
  emits: ['getPrevOffset', 'getNextOffset'],
  setup(props, { emit }) {
    const getNextOffset = (): void => {
      if (props.nextButtonDisabled) {
        return
      }

      emit('getNextOffset')
    }

    const getPrevOffset = (): void => {
      if (props.prevButtonDisabled) {
        return
      }

      emit('getPrevOffset')
    }

    return {
      getPrevOffset,
      getNextOffset,
    }
  },
})
</script>

<style lang="scss" scoped>
@import '@/styles/variables';
.pagination-offset-button-container {
  display: flex;
  .pagination-button.k-button {
    width: 34px;
    height: 34px;
    color: var(--grey-500);
    border: 1px solid var(--grey-300);
    background-color: white;
    border-radius: 4px;
    margin: 0 6px;
    padding: 6px;
    &:focus,
    &:hover {
      color: var(--blue-500);
      border-color: var(--blue-500);
      border-radius: 4px;
    }
    &.disabled:focus,
    &.disabled:hover {
      color: var(--black-45);
      border-color: var(--grey-200);
      box-shadow: none;
      cursor: default;
    }
    &.active {
      outline: none;
      color: var(--blue-500);
      border-color: var(--blue-200);
      border-radius: 4px;
      background-color: var(--blue-100);
    }
  }
}
</style>
