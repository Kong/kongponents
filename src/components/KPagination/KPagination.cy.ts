import { mount } from 'cypress/vue'
import KPagination from '@/components/KPagination/KPagination.vue'

describe('KPagination', () => {
  const myItems = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i']
  const pageSizes = [2, 4, 6]

  it('correctly renders props', () => {
    const currPage = 2
    mount(KPagination, {
      props: {
        totalCount: 9,
        pageSizes,
        currentPage: currPage,
        items: myItems,
        testMode: true,
      },
    })

    cy.getTestId('visible-items').should('contain.text', '3 to 4')
    cy.getTestId('visible-items').should('contain.text', 'of 9')
    cy.get('.pagination-button.active').should('contain.text', currPage + '')

    for (let i = 0; i < pageSizes.length; i++) {
      cy.get(`[data-testid="dropdown-item-trigger"][value="${pageSizes[i]}"]`).should('exist')
    }
  })

  it('goes to first page', () => {
    mount(KPagination, {
      props: {
        totalCount: 9,
        pageSizes: [2, 4, 6],
        items: myItems,
        testMode: true,
      },
    })

    cy.get('.pagination-button.active').should('contain.text', 1 + '')
    cy.getTestId('next-button').click()
    cy.getTestId('next-button').click()
    cy.get('.pagination-button.active').should('contain.text', 3 + '')
    cy.getTestId('page-1-button').click()
    cy.get('.pagination-button.active').should('contain.text', 1 + '')
  })

  it('goes to previous page', () => {
    mount(KPagination, {
      props: {
        totalCount: 9,
        pageSizes: [2, 4, 6],
        items: myItems,
        testMode: true,
      },
    })

    cy.get('.pagination-button.active').should('contain.text', 1 + '')
    cy.getTestId('next-button').click()
    cy.get('.pagination-button.active').should('contain.text', 2 + '')
    cy.getTestId('previous-button').click()
    cy.get('.pagination-button.active').should('contain.text', 1 + '')
  })

  it('can change page size', () => {
    mount(KPagination, {
      props: {
        totalCount: 9,
        pageSizes: [2, 4, 6],
        items: myItems,
        testMode: true,
      },
    })

    cy.getTestId('page-size-dropdown-trigger').should('contain.text', '2 items per page')
    cy.getTestId('page-size-dropdown-trigger').click({ multiple: true, force: true })
    cy.get('[data-testid="dropdown-item-trigger"][value="4"]').click({ multiple: true, force: true })
    cy.getTestId('page-size-dropdown-trigger').should('contain.text', '4 items per page')
  })
})
