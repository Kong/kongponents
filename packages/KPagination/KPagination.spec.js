import { mount } from '@vue/test-utils'
import KPagination from '@/KPagination/KPagination'

/**
 * ALL TESTS MUST USE testMode: true
 * We generate unique IDs for reference by aria properties. Test mode strips these out
 * allowing for successful snapshot verification.
 * propsData: {
 *   testMode: true
 * }
 */
describe('KPagination', () => {
  const myItems = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i']
  const pageSizes = [2, 4, 6]

  it('correctly renders props', () => {
    const currPage = 2
    const wrapper = mount(KPagination, {
      propsData: {
        totalCount: 9,
        pageSizes: pageSizes,
        currentPage: currPage,
        items: myItems,
        testMode: true
      }
    })

    expect(wrapper.find('[data-testid="visible-items"]').html()).toEqual(expect.stringContaining('3 to 4'))
    expect(wrapper.find('[data-testid="visible-items"]').html()).toEqual(expect.stringContaining('of 9'))
    expect(wrapper.find('.pagination-button.active a').html()).toEqual(expect.stringContaining(currPage + ''))

    for (let i = 0; i < pageSizes.length; i++) {
      expect(wrapper.find(`[data-testid="k-select-item-${pageSizes[i]}"]`).exists()).toBe(true)
    }

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('goes to first page', () => {
    const wrapper = mount(KPagination, {
      propsData: {
        totalCount: 9,
        pageSizes: [2, 4, 6],
        items: myItems,
        testMode: true
      }
    })

    expect(wrapper.find('.pagination-button.active').html()).toEqual(expect.stringContaining(1 + ''))
    wrapper.find('[data-testid="next-btn"] a').trigger('click')
    wrapper.find('[data-testid="next-btn"] a').trigger('click')
    expect(wrapper.find('.pagination-button.active').html()).toEqual(expect.stringContaining(3 + ''))
    wrapper.find('[data-testid="page-1-btn"] a').trigger('click')
    expect(wrapper.find('.pagination-button.active').html()).toEqual(expect.stringContaining(1 + ''))
  })

  it('goes to previous page', () => {
    const wrapper = mount(KPagination, {
      propsData: {
        totalCount: 9,
        pageSizes: [2, 4, 6],
        items: myItems,
        testMode: true
      }
    })

    expect(wrapper.find('.pagination-button.active').html()).toEqual(expect.stringContaining(1 + ''))
    wrapper.find('[data-testid="next-btn"] a').trigger('click')
    expect(wrapper.find('.pagination-button.active').html()).toEqual(expect.stringContaining(2 + ''))
    wrapper.find('[data-testid="prev-btn"] a').trigger('click')
    expect(wrapper.find('.pagination-button.active').html()).toEqual(expect.stringContaining(1 + ''))
  })

  it('can change page size', () => {
    const wrapper = mount(KPagination, {
      propsData: {
        totalCount: 9,
        pageSizes: [2, 4, 6],
        items: myItems,
        testMode: true
      }
    })

    expect(wrapper.find('[data-testid="k-select-input"] .k-button').html()).toEqual(expect.stringContaining('2 items per page'))
    wrapper.find('[data-testid="k-select-input"] button').trigger('click')
    wrapper.find('[data-testid="k-select-item-4"] button').trigger('click')
    expect(wrapper.find('[data-testid="k-select-input"] .k-button').html()).toEqual(expect.stringContaining('4 items per page'))
  })

  it('allows disabling page jump', () => {
    const wrapper = mount(KPagination, {
      propsData: {
        totalCount: 9,
        pageSizes: [2, 4, 6],
        items: myItems,
        disablePageJump: true,
        testMode: true
      }
    })

    expect(wrapper.find('[data-testid="page-1-btn"]').exists()).toBe(false)
    expect(wrapper.find('[data-testid="prev-btn"]').exists()).toBe(true)
    expect(wrapper.find('[data-testid="next-btn"]').exists()).toBe(true)
  })

  it('matches snapshot', () => {
    const wrapper = mount(KPagination, {
      propsData: {
        totalCount: 100,
        pageSizes: [10, 20, 30],
        neighbors: 2,
        testMode: true
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
