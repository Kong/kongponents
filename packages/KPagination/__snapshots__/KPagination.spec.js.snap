// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KPagination correctly renders props 1`] = `
<nav aria-label="Pagination Navigation" data-testid="k-pagination-container">
  <div class="card-pagination-bar"><span data-testid="visible-items" class="pagination-text"><span class="pagination-text-pages">3 to 4</span>
    of 9
    </span>
    <ul class="pagination-button-container">
      <li data-testid="prev-btn" class="pagination-button square"><a href="#" aria-label="Go to the previous page"><svg height="16px" width="16px" viewBox="0 0 16 14" role="img" class="kong-icon kong-icon-arrowLeft">
            <title>Back</title>
            <g>
              <!---->
              <!---->
              <path d="M12.17 8.75H3M6.75 5 3 8.75l3.75 3.75" fill="none" stroke="var(--blue-400)" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"></path>
            </g>
          </svg></a></li>
      <!---->
      <!---->
      <li data-testid="page-1-btn" class="pagination-button"><a aria-label="Go to page 1" href="#">
          1
        </a></li>
      <li data-testid="page-2-btn" class="pagination-button active"><a aria-label="Go to page 2" aria-current="page" href="#">
          2
        </a></li>
      <li data-testid="page-3-btn" class="pagination-button"><a aria-label="Go to page 3" href="#">
          3
        </a></li>
      <li data-testid="page-4-btn" class="pagination-button"><a aria-label="Go to page 4" href="#">
          4
        </a></li>
      <li data-testid="page-5-btn" class="pagination-button"><a aria-label="Go to page 5" href="#">
          5
        </a></li>
      <!---->
      <!---->
      <li data-testid="next-btn" class="pagination-button square"><a href="#" aria-label="Go to the next page"><svg height="16px" width="16px" viewBox="0 0 16 14" role="img" class="kong-icon kong-icon-arrowRight">
            <title>Forward</title>
            <g>
              <!---->
              <!---->
              <path d="M3 8.75h9.17M8.42 5l3.75 3.75-3.75 3.75" fill="none" stroke="var(--blue-400)" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"></path>
            </g>
          </svg></a></li>
    </ul> <span data-testid="page-size-dropdown" class="page-size-select"><div class="k-select" style="width: 200px;"><!----> <div id="test-select-id-1234" data-testid="k-select-selected-item"><!----> <div id="test-target-id-1234" aria-controls="test-popover-id-1234" role="button"><div id="test-select-input-id-1234" data-testid="k-select-input" role="listbox" class="k-select-button" style="position: relative;"><button type="button" class="k-button medium btn-link has-caret" id="test-select-text-id-1234" style="width: 200px;">2 items per page <svg height="24" width="24" viewBox="2 2 15 15" role="img" class="kong-icon kong-icon-chevronDown caret has-caret"><title>Expand</title>  <g><!----> <!----> <path d="m4.6 7 5.2 5L15 7" fill="none" stroke="#A3BBCC" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path></g></svg></button></div> <div id="test-popover-id-1234" role="region" class="k-popover k-select-popover mt-0 undefined k-select-pop-button hide-caret" style="display: none;" name="fade"><!----> <div class="k-popover-content"><ul class="k-select-list ma-0 pa-0"><div><div data-testid="k-select-item-2" class="k-select-item"><li role="option" class="d-block"><button value="2" class=""><span class="k-select-item-label mr-2">2</span> <span class="k-select-selected-icon-container"><!----></span></button></li>
  </div>
  <div data-testid="k-select-item-4" class="k-select-item">
    <li role="option" class="d-block"><button value="4" class=""><span class="k-select-item-label mr-2">4</span> <span class="k-select-selected-icon-container"><!----></span></button></li>
  </div>
  <div data-testid="k-select-item-6" class="k-select-item">
    <li role="option" class="d-block"><button value="6" class=""><span class="k-select-item-label mr-2">6</span> <span class="k-select-selected-icon-container"><!----></span></button></li>
  </div>
  </div>
  </ul>
  </div>
  <!---->
  </div>
  </div>
  </div>
  </div></span></div>
</nav>
`;

exports[`KPagination matches snapshot 1`] = `
<nav aria-label="Pagination Navigation" data-testid="k-pagination-container">
  <div class="card-pagination-bar"><span data-testid="visible-items" class="pagination-text"><span class="pagination-text-pages">1 to 10</span>
    of 100
    </span>
    <ul class="pagination-button-container">
      <li data-testid="prev-btn" class="pagination-button square disabled"><a href="#" aria-label="Go to the previous page"><svg height="16px" width="16px" viewBox="0 0 16 14" role="img" class="kong-icon kong-icon-arrowLeft">
            <title>Back</title>
            <g>
              <!---->
              <!---->
              <path d="M12.17 8.75H3M6.75 5 3 8.75l3.75 3.75" fill="none" stroke="var(--grey-500)" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"></path>
            </g>
          </svg></a></li>
      <!---->
      <!---->
      <li data-testid="page-1-btn" class="pagination-button active"><a aria-label="Go to page 1" aria-current="page" href="#">
          1
        </a></li>
      <li data-testid="page-2-btn" class="pagination-button"><a aria-label="Go to page 2" href="#">
          2
        </a></li>
      <li data-testid="page-3-btn" class="pagination-button"><a aria-label="Go to page 3" href="#">
          3
        </a></li>
      <li data-testid="page-4-btn" class="pagination-button"><a aria-label="Go to page 4" href="#">
          4
        </a></li>
      <li data-testid="page-5-btn" class="pagination-button"><a aria-label="Go to page 5" href="#">
          5
        </a></li>
      <li data-testid="page-6-btn" class="pagination-button"><a aria-label="Go to page 6" href="#">
          6
        </a></li>
      <li data-testid="page-7-btn" class="pagination-button"><a aria-label="Go to page 7" href="#">
          7
        </a></li>
      <li class="pagination-button placeholder">
        ...
      </li>
      <li class="pagination-button"><a href="#" aria-label="Go to the last page" data-testid="last-btn">
          10
        </a></li>
      <li data-testid="next-btn" class="pagination-button square"><a href="#" aria-label="Go to the next page"><svg height="16px" width="16px" viewBox="0 0 16 14" role="img" class="kong-icon kong-icon-arrowRight">
            <title>Forward</title>
            <g>
              <!---->
              <!---->
              <path d="M3 8.75h9.17M8.42 5l3.75 3.75-3.75 3.75" fill="none" stroke="var(--blue-400)" stroke-width="1.25" stroke-linecap="round" stroke-linejoin="round"></path>
            </g>
          </svg></a></li>
    </ul> <span data-testid="page-size-dropdown" class="page-size-select"><div class="k-select" style="width: 200px;"><!----> <div id="test-select-id-1234" data-testid="k-select-selected-item"><!----> <div id="test-target-id-1234" aria-controls="test-popover-id-1234" role="button"><div id="test-select-input-id-1234" data-testid="k-select-input" role="listbox" class="k-select-button" style="position: relative;"><button type="button" class="k-button medium btn-link has-caret" id="test-select-text-id-1234" style="width: 200px;">10 items per page <svg height="24" width="24" viewBox="2 2 15 15" role="img" class="kong-icon kong-icon-chevronDown caret has-caret"><title>Expand</title>  <g><!----> <!----> <path d="m4.6 7 5.2 5L15 7" fill="none" stroke="#A3BBCC" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path></g></svg></button></div> <div id="test-popover-id-1234" role="region" class="k-popover k-select-popover mt-0 undefined k-select-pop-button hide-caret" style="display: none;" name="fade"><!----> <div class="k-popover-content"><ul class="k-select-list ma-0 pa-0"><div><div data-testid="k-select-item-10" class="k-select-item"><li role="option" class="d-block"><button value="10" class=""><span class="k-select-item-label mr-2">10</span> <span class="k-select-selected-icon-container"><!----></span></button></li>
  </div>
  <div data-testid="k-select-item-20" class="k-select-item">
    <li role="option" class="d-block"><button value="20" class=""><span class="k-select-item-label mr-2">20</span> <span class="k-select-selected-icon-container"><!----></span></button></li>
  </div>
  <div data-testid="k-select-item-30" class="k-select-item">
    <li role="option" class="d-block"><button value="30" class=""><span class="k-select-item-label mr-2">30</span> <span class="k-select-selected-icon-container"><!----></span></button></li>
  </div>
  </div>
  </ul>
  </div>
  <!---->
  </div>
  </div>
  </div>
  </div></span></div>
</nav>
`;
